### PR TITLE
Add Allowed OAuth Tenants

### PR DESCRIPTION
**Pull request by vincent-stytch**

### CODE REVIEW

File: lib/b2c/m2m.ts [hunk 1]
```diff
@@ -241,19 +241,19 @@ export class M2M {
...
+                     * You may also override the default scope authorization function to implement custom authorization logic.
+                     *
+                     * @param data {@link AuthenticateTokenRequest}
+                     * @param scopeAuthorizationFunc {@link ScopeAuthorizationFunc} - A function that checks if the token has the required scopes. 
+                       The default function assumes scopes are either direct string matches or written in the form "action:resource". See the 
+                       documentation for {@link performAuthorizationCheck} for more information.
+                     * @async
+                     * @returns {@link AuthenticateTokenResponse}
+                     * @throws {ClientError} when token can not be authenticated
+                     */
```
Comment (logan-stytch) [inline lib/b2c/m2m.ts:256]: Hmmm... looks like codegen is forcing this section to slowly march right off the page. I'll file a task to look into why it's doing this.

**Review by logan-stytch — APPROVED** [review]

File: lib/b2c/m2m.ts (hunk 1 above)
Comment (logan-stytch) [reply lib/b2c/m2m.ts:256]: BACK-3893

File: dist/b2c/m2m.js [hunk 2]
```diff
@@ -97,19 +97,19 @@ class M2M {
...
+                     * You may also override the default scope authorization function to implement custom authorization logic.
+                     *
+                     * @param data {@link AuthenticateTokenRequest}
+                     * @param scopeAuthorizationFunc {@link ScopeAuthorizationFunc} - A function that checks if the token has the required scopes. 
+                       The default function assumes scopes are either direct string matches or written in the form "action:resource". See the 
+                       documentation for {@link performAuthorizationCheck} for more information.
+                     * @async
+                     * @returns {@link AuthenticateTokenResponse}
+                     * @throws {ClientError} when token can not be authenticated
+                     */
```
Comment (vincent-stytch) [inline dist/b2c/m2m.js:112]: @logan-stytch same thing happening here

Comment (logan-stytch) [reply dist/b2c/m2m.js:112]: Yep, this is just generated from the `.ts`, so that tracks

**Pull request merged**